File: Problems/2997. Minimum Number of Operations to Make Array XOR Equal to K/minimum-number-of-operations-to-make-array-xor-equal-to-k.ts
//Solution 1: 60ms 62.00 MB
function minOperations(nums: number[], k: number): number {
  let xorSum = k;
  for (const num of nums) {
    xorSum ^= num;
  }

  let minOperationCount = 0;
  while (xorSum > 0) {
    minOperationCount += xorSum & 1;
    xorSum >>= 1;
  }
  return minOperationCount;
}

File: Problems/2997. Minimum Number of Operations to Make Array XOR Equal to K/minimum-number-of-operations-to-make-array-xor-equal-to-k.java
//Solution 1: 1ms 56.74 MB
class Solution {
  public int minOperations(int[] nums, int k) {
    for (int num : nums) {
      k ^= num;
    }

    int minOperationCount = 0;
    while (k > 0) {
      minOperationCount += k & 1;
      k >>= 1;
    }

    return minOperationCount;
  }
}